.github/actions/build-rpm/action.yml

```diff
@@ -0,0 +1,148 @@
+                      --privileged=true
+                      -e DAOS_FULLNAME="${{ inputs.daos_full_name }}"
+                      -e DAOS_EMAIL="${{ inputs.daos_email }}"
+                      -e DISTRO_VERSION="${{ steps.build-vars.outputs.distro_version }}"
+                      -e STAGE_NAME="Build RPM on ${{ steps.build-vars.outputs.build_name }}"
+                      -e CHROOT_NAME="${{ steps.build-vars.outputs.chroot_name}}"
+                      -e ARTIFACTORY_URL="${{ inputs.artifactory_url}}"
+                      -e REPO_FILE_URL="${{ inputs.repo_file_url }}"
+                      -e JENKINS_URL=${{ inputs.jenkins_url }}
+                      -e TARGET="${{ inputs.distro }}"
+                      -e COVFN_DISABLED="${{ inputs.covfn_disabled }}"
+                      mock-build ci/rpm/build.sh
+      shell: bash
+    - name: Build RPM failure log
+      id: build-rpm-fail-log
+      continue-on-error: true
+      if: steps.build-rpm.outcome != 'success'
+      run: cat mock_result/root.log;
+            cat mock_result/build.log
+      shell: bash
+    - name: Save RPM build logs
+      continue-on-error: true
+      uses: actions/upload-artifact@v4
+      with:
+        name: ${{ steps.build-vars.outputs.stage_name }} logs
+        path: |
+          mock_result/root.log
+          mock_result/build.log
+    - name: Create lastBuild and lastSuccessfulBuild symlinks
+      if: steps.build-rpm.outcome == 'success'
+      run: . ci/gha_functions.sh;
+            mkdir -p ${REPO_PATH};
+            rm -f ${REPO_PATH}last{,Successful}Build;
+            ln -s ${{ github.run_number }} ${REPO_PATH}lastBuild;
+            ln -s ${{ github.run_number }} ${REPO_PATH}lastSuccessfulBuild
+      shell: bash
+    - name: Create repo
+      id: create-repo
+      if: steps.build-rpm.outcome == 'success'
+      continue-on-error: true
+      run: ci/rpm/create_repo.sh
+      shell: bash
+      env:
+        TARGET: ${{ inputs.distro }}
+    - name: Test repo
+      id: test-repo
+      if: steps.create-repo.outcome == 'success'
+      continue-on-error: true
+      run: . ci/gha_functions.sh;
+            dnf --disablerepo=\* --repofrompath testrepo,file://${REPO_FILE} repoquery -a
+      shell: bash
+      env:
+        # yamllint disable-line rule:line-length
+        REPO_FILE: ${{ inputs.repo_path }}${{ github.run_number }}/artifact/artifacts/${{ inputs.distro }}
+    - name: Remove lastSuccessfulBuild link and exit failure
+      if: steps.test-repo.outcome != 'success'
+      run: rm -f ${{ inputs.repo_path }}lastSuccessfulBuild;
+            exit 1
+      shell: bash
+    - name: Publish RPMs
+      uses: actions/upload-artifact@v4
+      with:
+        # yamllint disable-line rule:line-length
+        name: ${{ steps.build-vars.outputs.distro_name }} ${{ steps.build-vars.outputs.distro_version }} RPM repository
+        path: ${{ inputs.REPO_PATH}}${{ github.run_number }}/artifact/artifacts/${{ inputs.distro }}
```

Comment:
You can't do this.  Per SDL, we are not allowed to distribute unsigned binaries.

Reply:
This is the same code as https://github.com/daos-stack/daos/blob/master/.github/workflows/rpm-build-and-test.yml#L301-L305.  Doesn't it just add the RPMs to Artifactory so that the functional test stages can install those RPMs?

Reply:
> This is the same code as [`master`/.github/workflows/rpm-build-and-test.yml#L301-L305](https://github.com/daos-stack/daos/blob/master/.github/workflows/rpm-build-and-test.yml?rgh-link-date=2024-10-21T18%3A50%3A41Z#L301-L305).

Which is being replaced by https://github.com/daos-stack/actions-lib/pull/1/files#diff-46a90c32d59a22431144045d82ce732bc5a498ea20229b697bc2fd2154899662R423-R428 and https://github.com/daos-stack/daos/pull/14177/files#diff-681a2eb462568f244d2fd51eff96c3d9945e33dec02ee489c680a742a0bdcef4R79-R85 which you can see comments out that upload.

> Doesn't it just add the RPMs to Artifactory so that the functional test stages can install those RPMs?

No.  That uploads them to the artifacts in GitHub, where anyone can download them.  This is the part that puts us off-side of the SDL.

Moreover, we don't put RPMs from PRs into Artifactory.  With Jenkins, we install them from the artifacts of the Jenkins job itself and in GHA they are stored in this location https://github.com/daos-stack/daos/pull/14177/files#diff-681a2eb462568f244d2fd51eff96c3d9945e33dec02ee489c680a742a0bdcef4R79-R85 by [ci/rpm/create_repo.sh](https://github.com/daos-stack/daos/blob/pahender/DAOS-16285_master2/ci/rpm/create_repo.sh).